cmd/argo/commands/cron/lint.go

```diff
@@ -2,80 +2,68 @@ package cron
 
 import (
 	"fmt"
-	"os"
-	"path/filepath"
 
-	"github.com/argoproj/pkg/errors"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 
 	"github.com/argoproj/argo/cmd/argo/commands/client"
-	cronworkflowpkg "github.com/argoproj/argo/pkg/apiclient/cronworkflow"
+	"github.com/argoproj/argo/pkg/apiclient/cronworkflow"
+	"github.com/argoproj/argo/workflow/templateresolution"
 	"github.com/argoproj/argo/workflow/validate"
 )
 
 func NewLintCommand() *cobra.Command {
 	var (
-		strict bool
+		strict  bool
+		offline bool
 	)
 	var command = &cobra.Command{
 		Use:   "lint FILE...",
-		Short: "validate files or directories of cron workflow manifests",
+		Short: "validate files or directories of CronWorkflow manifests",
 		Run: func(cmd *cobra.Command, args []string) {
-			ctx, apiClient := client.NewAPIClient()
-			serviceClient := apiClient.NewCronWorkflowServiceClient()
-			namespace := client.Namespace()
+			resources, err := validate.ParseResourcesFromFiles(args, strict)
+			if err != nil {
+				log.Fatal(err)
+			}
 
-			lint := func(file string) error {
-				wfs, err := validate.ParseCronWorkflowsFromFile(file, strict)
-				if err != nil {
-					return err
-				}
-				if len(wfs) == 0 {
-					return fmt.Errorf("there was nothing to validate")
-				}
-				for _, wf := range wfs {
-					_, err := serviceClient.LintCronWorkflow(ctx, &cronworkflowpkg.LintCronWorkflowRequest{Namespace: namespace, CronWorkflow: &wf})
+			invalid := false
+
+			if offline {
+				templateGetter := templateresolution.WrapWorkflowTemplateList(resources.WorkflowTemplates)
+				clusterTemplateGetter := templateresolution.WrapClusterWorkflowTemplateList(resources.ClusterWorkflowTemplates)
+
+				for _, cron := range resources.CronWorkflows {
+					conditions, err := validate.ValidateCronWorkflow(templateGetter, clusterTemplateGetter, &cron, true)
 					if err != nil {
-						return err
+						log.Errorf("Error in CronWorkflow %s: %s", cron.Name, err)
+						invalid = true
+					}
+					for _, condition := range *conditions {
+						log.Warnf("Warning in CronWorkflow %s: %s", cron.Name, condition.Message)
 					}
 				}
-				fmt.Printf("%s is valid\n", file)
```

Comment:
Should I add back this print (and similar ones in the other `lint` commands)? I can see it both ways: sometimes it's nice to see all the positive validation messages, and other times it's nice to cut the noise and only see text when something has gone wrong.

Reply:
I don't think these should be removed, as some of our users might be expecting them in scripts/CIs